compiler/qsc_frontend/src/compile.rs

```diff
@@ -175,6 +183,9 @@ pub type SourceName = Arc<str>;
 
 pub type SourceContents = Arc<str>;
 
+// the arc<str> is only `None` for the legacy stdlib, core, and an interpreter special case
```

Comment:
Are the stdlib and core considered legacy after these changes? What makes them legacy?